lib/ukstore/store.c

```diff
@@ -34,6 +34,90 @@
 #include <errno.h>
 
 
+
+int uk_store_set_u8(struct uk_store_entry *e, __u8 *val)
+{
+	return e->set.u8(*val);
+}
+int uk_store_set_s8(struct uk_store_entry *e, __s8 *val)
+{
+	return e->set.s8(*val);
+}
+int uk_store_set_u16(struct uk_store_entry *e, __u16 *val)
+{
+	return e->set.u16(*val);
+}
+int uk_store_set_s16(struct uk_store_entry *e, __s16 *val)
+{
+	return e->set.s16(*val);
+}
+int uk_store_set_u32(struct uk_store_entry *e, __u32 *val)
+{
+	return e->set.u32(*val);
+}
+int uk_store_set_s32(struct uk_store_entry *e, __s32 *val)
+{
+	return e->set.s32(*val);
+}
+int uk_store_set_u64(struct uk_store_entry *e, __u64 *val)
+{
+	return e->set.u64(*val);
+}
+int uk_store_set_s64(struct uk_store_entry *e, __s64 *val)
+{
+	return e->set.s64(*val);
+}
+int uk_store_set_uptr(struct uk_store_entry *e, __uptr *val)
+{
+	return e->set.uptr(*val);
+}
+
+int uk_store_get_u8(struct uk_store_entry *e, __u8 *out)
+{
+	*out = e->get.u8();
+	return 0;
+}
+int uk_store_get_s8(struct uk_store_entry *e, __s8 *out)
```

Comment:
Similar to the setters you could do some checks that the target type is big enough to take the number.

``` c
int uk_store_get_s8(struct uk_store_entry *e, __s8 *out)
{
   int ret;

   UK_ASSERT(e);

   switch (e->type) {
   /* ... */

   case s64:
      __s64 val;
      ret = e->get.s64(&val);
      if (ret < 0)
         return ret;
      if (unlikely((val > __S8_MAX) || (val < __S8_MIN)))
         return -ERANGE;
      *out = (__s8) val;
      return ret;

   /* ... */
}